### PR TITLE
[Model Monitor] - Handle no data within model monitor components

### PR DESCRIPTION
**Pull request by mastloui-msft**

### CODE REVIEW

File: assets/model_monitoring/components/src/data_quality_compute_metrics/run.py [hunk 1]
```diff
@@ -5,7 +5,11 @@
 
 import argparse
 from pyspark.sql.functions import split, trim, col
-from shared_utilities.io_utils import read_mltable_in_spark, save_spark_df_as_mltable
+from shared_utilities.io_utils import (
+    try_read_mltable_in_spark_with_warning,
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/data_quality_compute_metrics/run.py:9]: nit: sort

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py [hunk 2]
```diff
@@ -87,55 +86,65 @@ def mdc_preprocessor(
...
 
     # Read mltable from preprocessed_data
-    df = read_mltable_in_spark(mltable_path=des_path)
+    df = try_read_mltable_in_spark(des_path, "preprocessed_data")
 
-    if df.count() == 0:
-        raise Exception(
-            "The window for this current run contains no data. "
+    if not df:
+        print(" ''.")
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:92]: Message?

File: assets/model_monitoring/components/src/model_monitor_compute_histogram/run.py [hunk 3]
```diff
@@ -11,7 +11,12 @@
     StringType,
     DoubleType,
 )
-from shared_utilities.io_utils import init_spark, read_mltable_in_spark, save_spark_df_as_mltable
+from shared_utilities.io_utils import (
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/model_monitor_compute_histogram/run.py:14]: nit: sort

File: assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py [hunk 4]
```diff
@@ -8,7 +8,7 @@
 from pyspark.sql.functions import col, udf, sum
 from pyspark.sql.types import IntegerType
 from shared_utilities.io_utils import (
-    read_mltable_in_spark,
+    try_read_mltable_in_spark_with_warning,
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py:11]: nit: sort

File: assets/model_monitoring/components/src/model_monitor_compute_histogram/run.py [hunk 5]
```diff
@@ -35,13 +40,16 @@ def run():
...
 
-    histogram_buckets = None
-    try:
-        histogram_buckets = read_mltable_in_spark(args.histogram_buckets)
-    except Exception:
-        print("No histogram buckets detected.")
+    histogram_buckets = try_read_mltable_in_spark(
+        args.histogram_buckets, "histogram_buckets"
+    )
+    if not histogram_buckets:
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/model_monitor_compute_histogram/run.py:52]: I see you removed a line of logging for no histogram. Is this intented?

**Review by yetamsft — APPROVED** [review]

File: assets/model_monitoring/components/src/model_monitor_compute_histogram/run.py (hunk 5 above)
Comment (mastloui-msft) [reply assets/model_monitoring/components/src/model_monitor_compute_histogram/run.py:52]: I'll add back!

File: assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py (hunk 4 above)
Comment (mastloui-msft) [reply assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py:11]: will defer!

File: assets/model_monitoring/components/src/model_monitor_compute_histogram/run.py (hunk 3 above)
Comment (mastloui-msft) [reply assets/model_monitoring/components/src/model_monitor_compute_histogram/run.py:14]: will defer!

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py (hunk 2 above)
Comment (mastloui-msft) [reply assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:92]: yeap, I'll need to add a message, will defer to the next one to get this big PR in

File: assets/model_monitoring/components/src/data_quality_compute_metrics/run.py (hunk 1 above)
Comment (mastloui-msft) [reply assets/model_monitoring/components/src/data_quality_compute_metrics/run.py:9]: will defer!

**Pull request merged**